docs/shadow_config_spec.md

```diff
@@ -670,7 +670,10 @@ produce hosts with hostnames `host1` and `host2`.
 *Required*  
 Type: Array
 
-Virtual software processes that the host will run.
+Virtual software processes that the host will run. PIDs are assigned from 1000
+on each host, in the order that they appear in this process list. e.g. this
+property can be used to cleanly shut down a process by scheduling a `/bin/kill`
+process to send a shutdown signal (e.g. `SIGTERM` or `SIGINT`) at the desired time.
```

Comment:
Hmm, it looks like I was wrong about the order that pids are assigned, and shadow assigns pids to processes before the simulation even starts. Is this something we should change to be more like Linux? Should we sort the processes by their start time before assigning pids?

While I think that assigning pids based on when the processes start is more intuitive (and follows Linux), it make the pids less easy to predict, since for example Shadow increments the pid number when new threads are created, and in the future for each fork.

Reply:
Yeah, this was a little surprising to me, too. I'm inclined to leave it as is though since we plan to remove this guarantee once we support `fork` and/or `/proc`, and until we do support `fork` the current behavior is a little easier to work with.

Reply:
Actually, if we assigned PIDs when the process starts, it'd be pretty difficult to predict them in the presence of threaded programs, since thread IDs are also PIDs. e.g. if the first process that started had 2 threads, then the next process that started would get pid 1002, not 1001.